frontend/src/pages/TracesExplorer/Filter/Filter.tsx

```diff
@@ -0,0 +1,247 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import './Filter.styles.scss';
+
+import {
+	FilterOutlined,
+	SyncOutlined,
+	VerticalAlignTopOutlined,
+} from '@ant-design/icons';
+import { Button, Flex, Tooltip, Typography } from 'antd';
+import { getMs } from 'container/Trace/Filters/Panel/PanelBody/Duration/util';
+import { useGetCompositeQueryParam } from 'hooks/queryBuilder/useGetCompositeQueryParam';
+import { useQueryBuilder } from 'hooks/queryBuilder/useQueryBuilder';
+import { isArray, isEqual } from 'lodash-es';
+import {
+	Dispatch,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+import { BaseAutocompleteData } from 'types/api/queryBuilder/queryAutocompleteResponse';
+import { Query, TagFilterItem } from 'types/api/queryBuilder/queryBuilderData';
+import { v4 as uuid } from 'uuid';
+
+import {
+	AllTraceFilterKeys,
+	AllTraceFilterKeyValue,
+	AllTraceFilterOptions,
+	FilterType,
+	HandleRunProps,
+	unionTagFilterItems,
+} from './filterUtils';
+import { Section } from './Section';
+
+interface FilterProps {
+	setOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+export function Filter(props: FilterProps): JSX.Element {
+	const { setOpen } = props;
+	const [selectedFilters, setSelectedFilters] = useState<
+		Record<
+			AllTraceFilterKeys,
+			{ values: string[] | string; keys: BaseAutocompleteData }
+		>
+	>();
+
+	const { currentQuery, redirectWithQueryBuilderData } = useQueryBuilder();
+	const compositeQuery = useGetCompositeQueryParam();
```

Comment:
NIT: better to use `stagedQuery` here as it will be in sync with query builder state